src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -77,27 +77,27 @@ public static ReadOnlySpan<byte> AsBytes<T>(ReadOnlySpan<T> span)
         /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to the location where the 0th element
         /// would have been stored. Such a reference may or may not be null. It can be used for pinning but must never be dereferenced.
         /// </summary>
-        public static ref T GetReference<T>([LifetimeAnnotation(false, true)] Span<T> span) => ref span._reference.Value;
+        public static ref T GetReference<T>(Span<T> span) => ref Unsafe.AsRef<T>(in span._reference);
 
         /// <summary>
         /// Returns a reference to the 0th element of the ReadOnlySpan. If the ReadOnlySpan is empty, returns a reference to the location where the 0th element
         /// would have been stored. Such a reference may or may not be null. It can be used for pinning but must never be dereferenced.
         /// </summary>
-        public static ref T GetReference<T>([LifetimeAnnotation(false, true)] ReadOnlySpan<T> span) => ref span._reference.Value;
+        public static ref T GetReference<T>(ReadOnlySpan<T> span) => ref Unsafe.AsRef<T>(in span._reference);
 
         /// <summary>
         /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to fake non-null pointer. Such a reference can be used
         /// for pinning but must never be dereferenced. This is useful for interop with methods that do not accept null pointers for zero-sized buffers.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static unsafe ref T GetNonNullPinnableReference<T>(Span<T> span) => ref (span.Length != 0) ? ref span._reference.Value : ref Unsafe.AsRef<T>((void*)1);
+        internal static unsafe ref T GetNonNullPinnableReference<T>(scoped Span<T> span) => ref (span.Length != 0) ? ref Unsafe.AsRef<T>(in span._reference) : ref Unsafe.AsRef<T>((void*)1);
```

Comment:
Just curious, how come this and the one below need to be `scoped`? Shouldn't it be fine to leave they unscoped since they're meant to give you a pin able reference to the same span you can pin, not a reference you'd use to escape the span content out of scope? 🤔

Reply:
I expect it shouldn't be scoped, just as we decided MemoryMarshal.GetReference shouldn't cheat.

Reply:
Yep. This is a leftover from a previous point in time with an earlier compiler.